packages/semantic-release-commit-rules/lib/defaults.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * @fileoverview Commit message configuration definition and default configuration
+ */
+
+/**
+ * Commit message scope configuration
+ * @typedef {object} scopeConfig
+ * @property {string} scope - commit scope to test against. Can use `*` as a wildcard
+ *    (e.g. MYSCOPE-* will find commits that have a scope that starts with MYSCOPE-)
+ * @property {string} release - release type to be generated when using commit type;
+ *    Available options: 'minor', 'patch', 'major'
+ * @property {string} cleanupLogs - string that replaces the value of `scope` when generating
+ *    GitHub release notes (e.g. For scope definition `MYSCOPE-*`, using '' will remove `MYSCOPE-`
+ *    from displaying)
+ */
+
+/**
+ * Commit message type configuration
+ * @typedef {object} typeConfig
+ * @property {string} release - release type to be generated when using commit type;
+ *    Available options: 'minor', 'patch', 'major', 'false'
+ * @property {string} title - Title used in GitHub's release notes when grouping commits with a similar type
+ * @property {boolean} [removeFromLogs=false] - turns off GitHub release note generate for commits with the defined type
+ * @property {scopeConfig[]} scopes - additional configurations for specific scopes when used with the specified type
+ */
+
+/**
+ * Commit message configurations for a package that are used to determine
+ * Semantic Release version type and GitHub release notes
+ * @typedef {object} commitConfigs
+ * @property {typeConfig} * - property name is the commit-message type and the value is a typeConfig
+ *      commitConfigs can have any number of properties as long as each follows this format; Using
+ *      the keyword `none` as the property name with a `scopes` array allows the user to define type
+ *      agnostic scope rules
+ */
+
+/**
+ * Default commit message configurations
+ * @typedef {commitConfigs} defaultCommitConfig
+ * @property {typeConfig} feat - configurations for commits that have type `feat`
+ * @property {typeConfig} build - configurations for commits that have type `build`
+ * @property {typeConfig} perf - configurations for commits that have type `perf`
+ * @property {typeConfig} fix - configurations for commits that have type `fix`
+ * @property {typeConfig} test - configurations for commits that have type `test`
+ * @property {typeConfig} ci - configurations for commits that have type `ci`
+ * @property {typeConfig} docs - configurations for commits that have type `docs`
+ * @property {typeConfig} style - configurations for commits that have type `style`
+ * @property {typeConfig} refactor - configurations for commits that have type `refactor`
+ * @property {typeConfig} none - scope configurations for commits of any type
+ */
+const defaultConfig = {
+  feat: {
```

Comment:
suggestion: overall this seems like it's building on top of the `releaserules` structure, which varies from this object only that the keys are properties and the whole thing is an array. Seems like this might be more familiar to users if this object was closer to the end-result in it's stored config by being an array of `typeConfig`s

This would reduce the complexity of `getReleaseRules` as well, only needing to account for the non-traditional properties e.g. `scopes`

Would also remove the need to jsdoc each rule for `defaultCommitConfig`. That understanding should come from the type configs themselves via `type.title`

Reply:
@scottnath At this time I don't want to change this object. I had tried multiple different structures when working on this in the sandbox and this is what worked best for the entire functionality of tool.  Once we have more of the functionality available for review, we can re-take a look at it if you'd like.